### PR TITLE
Feat game flavors

### LINKED ISSUE #46: Refactor Game Flavor
Instead of maintaining a list of several flavors of the same game (301, 501, 501-double-out,...), the flavor (also called GameStyle) should be :

- The Game type (x01, Cricket, ...)
- The rules and options description
- The options typology (name and type of each option element)

Then the UI should dynamically display a form corresponding to those options and then send to the server for game creation. If players want a 703 game based on 301 rules, so they can.

### PR DESCRIPTION
**Pull request by ggerbaud**
Fix #46

### CODE REVIEW

**Comment by mathieu-pousse** [comment]
Hum, look's interesting but all the message translations comes from everywhere (from each game program file)
The i18n coming from the backend is also an odd pattern i think

couldn't we use something in the frontend ?

File: game/cricket.go [hunk 1]
```diff
@@ -7,6 +7,7 @@ import (
 
 	log "github.com/Sirupsen/logrus"
 	"github.com/gocaine/go-dart/common"
+	"github.com/pkg/errors"
```
Comment (ggerbaud) [inline game/cricket.go]: fixed

File: game/countup.go [hunk 2]
```diff
@@ -6,6 +6,7 @@ import (
 
 	log "github.com/Sirupsen/logrus"
 	"github.com/gocaine/go-dart/common"
+	"github.com/pkg/errors"
```
Comment (ggerbaud) [inline game/countup.go]: fixed

**Comment by erwannT** [comment]
LGTM

**Pull request merged**